src/main/kotlin/com/example/passmanager/repositories/impl/PassRepositoryImpl.kt

```diff
@@ -0,0 +1,125 @@
+        val aggregation = newAggregation(
+            getPassesByOwnerAndPurchasedAfterPipeline(passOwnerId, afterDate).operations
+        )
+        val aggregationResults = mongoTemplate.aggregate<MongoPass>(aggregation, COLLECTION_NAME)
+        return aggregationResults.mappedResults
+    }
+
+    override fun findAllByPassOwnerId(passOwnerId: String): List<MongoPass> {
+        val pipelineStages = getPassesLookUpAggregationPipeline(passOwnerId).operations
+        val aggregationResults =
+            mongoTemplate.aggregate<ProjectionResult>(
+                newAggregation(pipelineStages),
+                PASS_OWNER_COLLECTION
+            )
+        return aggregationResults.uniqueMappedResult?.passes.orEmpty()
+    }
+
+    override fun findById(passId: String): MongoPass? {
+        return mongoTemplate.findById<MongoPass>(passId)
+    }
+
+    override fun insert(newPass: MongoPass): MongoPass {
+        return mongoTemplate.insert(newPass)
+    }
+
+    override fun save(pass: MongoPass): MongoPass {
+        return mongoTemplate.save(pass)
+    }
+
+    override fun deleteById(passId: String) {
+        mongoTemplate.remove<MongoPass>(query(where("_id").`is`(passId)))
+    }
+
+    override fun deleteAllByOwnerId(passOwnerId: String) {
+        mongoTemplate.remove<MongoPass>(query(where("passOwner._id").`is`(ObjectId(passOwnerId))))
+    }
+
+    override fun deleteByIdAndOwnerId(passId: String, passOwnerId: String): Boolean {
+        val deleteResult = mongoTemplate.remove<MongoPass>(
+            query(where("passOwner._id").`is`(ObjectId(passOwnerId)).andOperator(where("_id").`is`(passId)))
+        )
+        return deleteResult.wasAcknowledged()
+    }
+
+    override fun sumPurchasedAtAfterDate(passOwnerId: String, afterDate: LocalDate): BigDecimal {
+        val aggregation = newAggregation(
+            getPassesByOwnerAndPurchasedAfterPipeline(passOwnerId, afterDate)
+                .add(group().sum("purchasedFor").`as`("total"))
+                .add(project("total").andExclude("_id"))
+                .operations
+        )
+        val aggregationResults = mongoTemplate.aggregate<SumResult>(aggregation, COLLECTION_NAME)
+
+        return aggregationResults.uniqueMappedResult?.total ?: BigDecimal.ZERO
+    }
+
+    override fun getPassesPriceDistribution(passOwnerId: String): List<PriceDistribution> {
+        val aggregation = newAggregation(
+            match(where("passOwnerId").`is`(ObjectId(passOwnerId))),
+            lookup().from(PASS_TYPE_COLLECTION)
+                .localField("passTypeId")
+                .foreignField("_id")
+                .`as`("types"),
+            unwind("types"),
+            group("types.name").sum("types.price").`as`("spentForPassType"),
+            project("spentForPassType").and("_id").`as`("typeName").andExclude("_id")
+        )
+        val aggregationResults = mongoTemplate.aggregate<PriceDistribution>(aggregation, COLLECTION_NAME)
```

Comment:
You can use this, but that's up to you
```suggestion
        val aggregationResults = mongoTemplate.aggregate(aggregation, COLLECTION_NAME, PriceDistribution::class.java)
```